server/src/main/java/org/elasticsearch/action/admin/indices/refresh/UnpromotableShardRefreshRequest.java

```diff
@@ -23,8 +23,12 @@ public class UnpromotableShardRefreshRequest extends BroadcastUnpromotableReques
 
     private final long segmentGeneration;
 
-    public UnpromotableShardRefreshRequest(IndexShardRoutingTable indexShardRoutingTable, long segmentGeneration) {
-        super(indexShardRoutingTable);
+    public UnpromotableShardRefreshRequest(
+        IndexShardRoutingTable indexShardRoutingTable,
+        long segmentGeneration,
+        boolean failShardOnError
+    ) {
+        super(indexShardRoutingTable, failShardOnError);
```

Comment:
Why do we need this flag? Shouldn't we always fail the search shard if its refresh fails?

Reply:
The refresh api propagates an error and does not fail the shard.